src/test/kotlin/org/jmailen/gradle/kotlinter/functional/WithGradleTest.kt

```diff
@@ -0,0 +1,50 @@
+package org.jmailen.gradle.kotlinter.functional
+
+import org.gradle.internal.classpath.DefaultClassPath
+import org.gradle.testkit.runner.BuildResult
+import org.gradle.testkit.runner.GradleRunner
+import org.gradle.testkit.runner.internal.PluginUnderTestMetadataReading
+import org.junit.Rule
+import org.junit.rules.TemporaryFolder
+
+abstract class WithGradleTest {
+
+    @get:Rule
+    val testProjectDir = TemporaryFolder()
+
+    protected fun build(vararg args: String): BuildResult = gradleRunnerFor(*args).build()
+
+    protected fun buildAndFail(vararg args: String): BuildResult = gradleRunnerFor(*args).buildAndFail()
+
+    protected abstract fun gradleRunnerFor(vararg args: String): GradleRunner
+
+    abstract class Android : WithGradleTest() {
+
+        override fun gradleRunnerFor(vararg args: String): GradleRunner {
+            return defaultRunner(*args)
+                .withPluginClasspath()
+        }
+    }
+
+    abstract class Kotlin : WithGradleTest() {
+
+        override fun gradleRunnerFor(vararg args: String): GradleRunner {
+            val classpath = DefaultClassPath.of(PluginUnderTestMetadataReading.readImplementationClasspath()).asFiles
+            val androidDependencies = listOf(
+                ".*/com\\.android\\..*/.*".toRegex(),
+                ".*/androidx\\..*/.*".toRegex(),
+                ".*/com\\.google\\..*/.*".toRegex()
+            )
+            val noAndroid = classpath.filterNot { dependency -> androidDependencies.any { it.matches(dependency.path) } }
+
+            return defaultRunner(*args)
+                .withPluginClasspath(noAndroid)
```

Comment:
Ok, so here's my suggestion how to simulate the environment without android gradle plugin applied.
I'm removing all android depedencies from the classpath. The code with `dependOn` import fails as expected with 
```
java.lang.NoClassDefFoundError: com/android/build/gradle/internal/tasks/factory/TaskFactoryUtils
```

Reply:
I will say this feels a tiny bit fragile. My preferences would be (in a future PR at some point) introduce properly isolated test environments. Possible this would need to be done using two or more test sub-projects which only include kotlinter-gradle via buildscript api as to be more like real projects.

Reply:
Initially I thought about separating them by configuration, and there would be separate set of tasks for each environment (i.e. `testAndroid`, `testKotlin`, `estMultiplatform`). So they wouldn't be completely isolated (as they would still share `implementation` configuration), but that would allow us to have control over dependencies that are currently added to `compileOnly`.

However, having tests that introduce completely isolated enviroment sounds even better. I'll try to find some time over the weekend and figure something out.